lib/frontend/src/Obelisk/Frontend.hs

```diff
@@ -145,32 +146,36 @@ data FrontendMode = FrontendMode
 -- Selects FrontendMode based on platform; this doesn't work for jsaddle-warp
 runFrontend
   :: forall backendRoute route
-  .  Encoder Identity Identity (R (FullRoute backendRoute route)) PageName
+  .  Encoder (Either Text) Identity (R (FullRoute backendRoute route)) PageName
   -> Frontend (R route)
-  -> JSM ()
-runFrontend validFullEncoder frontend = do
-  let mode = FrontendMode
-        { _frontendMode_hydrate =
+  -> IO ()
+runFrontend fullRouteEncoder frontend =
+  case checkEncoder fullRouteEncoder of
+    Right validFullEncoder -> run $ do
+      let mode = FrontendMode
+            { _frontendMode_hydrate =
 #ifdef ghcjs_HOST_OS
-          True
+              True
 #else
-          False
+              False
 #endif
-        , _frontendMode_adjustRoute =
+            , _frontendMode_adjustRoute =
 #ifdef ghcjs_HOST_OS
-          False
+              False
 #else
-          True
+              True
 #endif
-        }
-  configs <- liftIO Lookup.getConfigs
-  when (_frontendMode_hydrate mode) removeHTMLConfigs
-  -- There's no fundamental reason that adjustRoute needs to control setting the
-  -- initial route and *also* the useHash parameter; that's why these are
-  -- separate here.  However, currently, they are always the same.
-  when (_frontendMode_adjustRoute mode) $ do
-    setInitialRoute $ _frontendMode_adjustRoute mode
-  runFrontendWithConfigsAndCurrentRoute mode configs validFullEncoder frontend
+            }
+      configs <- liftIO Lookup.getConfigs
+      when (_frontendMode_hydrate mode) removeHTMLConfigs
+      -- There's no fundamental reason that adjustRoute needs to control setting the
+      -- initial route and *also* the useHash parameter; that's why these are
+      -- separate here.  However, currently, they are always the same.
+      when (_frontendMode_adjustRoute mode) $ do
+        setInitialRoute $ _frontendMode_adjustRoute mode
+      runFrontendWithConfigsAndCurrentRoute mode configs validFullEncoder frontend
+    Left err ->
+      error $ "Route encoder check failed: " <> show err
```

Comment:
I still prefer having this check in the skeleton for two reasons

  1. It makes the error customizable and handling also somewhat customizable
  2. In order to actually *use* a route encoder, you usually have to run `checkEncoder` first anyway, so now we force users to have *two* places in the code where the `checkEncoder` happens.

What I think would be better is to put a `checkedFullRouteEncoder` in the skeleton that has already been checked and use that in `runFrontend`. Users can customize that if they want to but it also makes the actual `main` simpler.